rotkehlchen/chain/evm/accounting/interfaces.py

```diff
@@ -51,63 +60,194 @@ def reset(self) -> None:
+        events_to_consume = event.extra_data['withdrawal_events_num']
+        for idx in range(events_to_consume):
+            next_event = next(other_events, None)
+            if next_event is None:
+                log.debug(f'Could not consume event nr. {idx} for {event.counterparty} withdrawal')  # noqa: E501
+                return
+
+            if next_event.balance.amount == ZERO:
+                continue
+
+            usd_value = self._query_historical_price(pot=pot, event=next_event)
+            self._calculate_pnl(
+                pot=pot,
+                event=next_event,
+                usd_value=usd_value,
+            )
+            self._add_asset_change(
+                pot=pot,
+                method='acquisition',
+                event=next_event,
+            )
+
+    def process_deposit(
+            self,
+            pot: 'AccountingPot',
+            event: 'EvmEvent',
+            other_events: Iterator['EvmEvent'],
+    ) -> None:
+        """Process deposits to protocols and stores the USD value of the amount(s) deposited."""
+        if not (
+            event.extra_data is not None and
+            'deposit_events_num' in event.extra_data and
+            event.counterparty in {'curve', 'balancer-v1'}
+        ):
+            usd_value = self._query_historical_price(pot=pot, event=event)
+            self.amounts_deposited[event.asset] += Balance(
+                amount=event.balance.amount,
+                usd_value=usd_value,
             )
             return
 
-        # Consume the events
+        # it's curve or balancer protocol
+        events_to_consume = event.extra_data['deposit_events_num']
         for idx in range(events_to_consume):
             next_event = next(other_events, None)
             if next_event is None:
-                log.debug(f'Could not consume event nr. {idx} for {event.counterparty} deposit/withdrawal')  # noqa: E501
+                log.debug(f'Could not consume event nr. {idx} for {event.counterparty} deposit')  # noqa: E501
                 return
 
             if next_event.balance.amount == ZERO:
                 continue
 
-            pot.add_asset_change_event(
-                method=method,
-                event_type=AccountingEventType.TRANSACTION_EVENT,
-                notes=next_event.notes if next_event.notes else '',
-                location=next_event.location,
-                timestamp=next_event.get_timestamp_in_sec(),
-                asset=next_event.asset,
-                amount=next_event.balance.amount,
-                taxable=False,  # Deposits and withdrawals are not taxable
-                count_entire_amount_spend=False,
-                count_cost_basis_pnl=False,
-                extra_data={'tx_hash': next_event.serialized_event_identifier},
+            usd_value = self._query_historical_price(pot=pot, event=next_event)
+            # balancer special case handling of refunds which are not deposits
+            if (
```

Comment:
the code between the two functions seems to be almost the same with the exception of this part here and the key that you use. Why not abstract this both and make one function that can be used for both?

I believe you can add an statement with `if is_deposit` where `is_depost = 'deposit_events_num' in event.extra_data`

what do you think? The problem is that most of the logic is copied in both functions